### PR TITLE
Creation meetings reports

### PR DESCRIPTION
**Pull request by stahanna**
draft-create is just a draft

### CODE REVIEW

**Comment by stahanna** [comment]
Example of a file created:

# Gammapy Developer Meeting 
 * Tuesday, October 10, 2023, at 2 pm (CET) 
 * Gammapy Developer Meeting on Zoom (direct link on Slack) 
# Agenda

### PRs opened last week (less than 8 days ago): 
* [#4785](https://github.com/gammapy/gammapy/pull/4785) Use interpolation for dark matter mass - Stefan Fröse
* [#4783](https://github.com/gammapy/gammapy/pull/4783) Add EnergyDependentEstimator - Astro-Kirsty

### PRs merged last week (less than 8 days ago): 
* [#4779](https://github.com/gammapy/gammapy/pull/4779) Backport PR #4777 on branch v1.1.x (Bump actions/checkout from 3 to 4) - Lumberbot (aka Jack)
* [#4778](https://github.com/gammapy/gammapy/pull/4778) Backport PR #4777 on branch v1.0.x (Bump actions/checkout from 3 to 4) - Lumberbot (aka Jack)

### issues opened last week (less than 8 days ago): 
* [#4788](https://github.com/gammapy/gammapy/issues/4788) Expose undocumented new features from v1.1 and v1.0 - Régis Terrier
* [#4787](https://github.com/gammapy/gammapy/issues/4787) Developper documentation on metadata - REGEARD Maxime
* [#4786](https://github.com/gammapy/gammapy/issues/4786) Implement a split and merge scheme for Map - Axel Donath
* [#4785](https://github.com/gammapy/gammapy/pull/4785) Use interpolation for dark matter mass - Stefan Fröse
* [#4784](https://github.com/gammapy/gammapy/issues/4784) Adding a GTI.difference or GTI.remove_interval method - Régis Terrier
* [#4783](https://github.com/gammapy/gammapy/pull/4783) Add EnergyDependentEstimator - Astro-Kirsty
* [#4782](https://github.com/gammapy/gammapy/issues/4782) Do we need a container for phasogram ? - REGEARD Maxime
* [#4781](https://github.com/gammapy/gammapy/issues/4781) Nearest neighbour interpolation for mass of `PrimaryFlux` produces high fluctuations for ULs - Stefan Fröse
* [#4780](https://github.com/gammapy/gammapy/issues/4780) Support multiple time intervals in `gammapy.data.ObservationFilter` - Jean-Philippe Lenain

 report created at 10/10/2023, 14:57:52

File: create_dev_meetings_report.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import os
```
Comment (bkhelifi) [inline create_dev_meetings_report.py:1]: Add the licence line at the top of the file

File: draft_create.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import sys
```
Comment (bkhelifi) [inline draft_create.py:1]: Add the licence line at the top of the file

File: create_dev_meetings_report.py (hunk 1 above)
Comment (bkhelifi) [inline create_dev_meetings_report.py:1]: A small description of the code might be great, as well in the other python script

File: requirements.txt
Comment (bkhelifi) [inline requirements.txt:1]: This file is a new file in the top directory of gammapy. Is it how we should proceed? Should these data not placed in `environment-dev.yml'?

Comment (stahanna) [reply requirements.txt:1]: environment-dev.yml is for conda, isn't it? I don' know conda but I can make a file in this format (and will probably have to modify the dev_meeting_reports.yml to install the dependencies? I'm not sure about how to do it, but I will have a look)
There is only one .py file, the draft is to be ignored (just for tests). 
I will check how to add the license and will think about a short comment of the script. 
Thanks a lot for your remarks :)

Comment (registerrier) [reply requirements.txt:1]: @bkhelifi . This is supposed to go in `gammapy-meetings` to allow the scripts to run and is not meant to be part of a distribution. I think the `requirements.txt` is fine.

Comment (stahanna) [reply requirements.txt:1]: in this case the license line is not necessary either, or is it?

Comment (bkhelifi) [reply requirements.txt:1]: for requirements.txt, indeed. For the python files, I think that one should include it

File: create_dev_meetings_report.py (hunk 1 above)
Comment (stahanna) [reply create_dev_meetings_report.py:1]: something like this at the top of the create_dev_meetings_reports.py ? :  """
this script will run once a week (on Friday at 6 am? Time will be defined in the dev_meeting_reports.yml
- cron syntax) to create a Gammapy Developer Meeting Report containing pull requests which have been
opened or merged as well as issues opened in the last 7 (8?)  days
"""

Comment (stahanna) [reply create_dev_meetings_report.py:1]: the standard "# Licensed under a 3-clause BSD style license - see LICENSE.rst" I see in many other gammapy scripts?

Comment (bkhelifi) [reply create_dev_meetings_report.py:1]: yes, exactly! :)

**Comment by stahanna** [comment]
I commited the modified files

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+astropy==5.3.4
+bokeh==3.3.0
+matplotlib==3.8.0
+numpy==1.24.2
+PyGithub==2.1.1
+PyGithub==2.1.1
```
Comment (registerrier) [inline requirements.txt:6]: you can remove this line

**Review by registerrier — COMMENTED** [review]
Thanks @stahanna . This looks good. I have one small inline comment.

**Review by registerrier — COMMENTED** [review]
Thanks @stahanna ! This is nice. 
Could you just remove the draft_create.py file as it is no longer needed? An move the python script to a `scripts` directoy?

**Comment by stahanna** [comment]
I have deleted the draft and moved the create file to scripts

File: .github/workflows/dev_meeting_reports.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
+name : create meeting report
+on:
+  schedule:
+    - cron:  '0 16 * * 5' # create at 4 pm on Friday
```
Comment (AtreyeeS) [inline .github/workflows/dev_meeting_reports.yml]: Friday morning is better, no?

**Review by AtreyeeS — APPROVED** [review]
Thank you very much @stahanna ! Very useful!

File: .github/workflows/dev_meeting_reports.yml [hunk 5]
```diff
@@ -0,0 +1,33 @@
+name : create meeting report
+on:
+  schedule:
+    - cron:  '0 6 * * 5' # create at 4 pm on Friday
```
Comment (registerrier) [inline .github/workflows/dev_meeting_reports.yml:4]:
```suggestion
    - cron:  '0 6 * * 5' # create at 6 am on Friday
```

**Review by registerrier — APPROVED** [review]
Thanks @stahanna . I have removed the .keep file and temporarily put a different date for a test.
Merging now.

**Pull request merged**